src/api/db/data/20250710191133_backfill_status_on_bs_requests.rb

```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+class BackfillStatusOnBsRequests < ActiveRecord::Migration[7.2]
+  def up
+    BsRequest.where(status: nil).in_batches do |batch|
```

Comment:
This will probably take ages to finish. I would move it into a Rake task that we can run outside the deployment window.

Reply:
We don't need downtime for running these steps separately. We don't need to perfom this deployment in the deployment window...

Reply:
I'm not talking about the "maintenance window" I'm talking about the "deployment window". During the deployment window we push code to production, run migrations, and reload apache making the new code available. If the migrations take a lot of time, the reloading of the apache and the code switch is being delayed...

Reply:
In case of errors, you could also adjust the migrations and rollback or apply migrations.

> If the migrations take a lot of time, the reloading of the apache and the code switch is being delayed...

I agree that reloading apache after a heavy migration is delayed.

However, first running a heavy migration, and then reloading the apache after a delay isn't an issue in this case. These changes of code, migrations and data migrations are created and split in a way the changes can be deployed without downtime.

> Also, running the data migration by hand it's easier to react to errors, you can fix whatever you need and run it again.

In data migrations which require downtime, yes, I agree. This is not the case.

Reply:
No one is talking about downtime! I'm talking about having an increasing amount of time between the deployed code and the reloading of the services, while the application is running live. But whatever.

Reply:
> No one is talking about downtime! I'm talking about having an increasing amount of time between the deployed code and the reloading of the services, while the application is running live. But whatever.

Increasing... I would say huge amount of time.

I would like to insist, that if this data migration take even days, it doesn't require any downtime. In other words, the application can still run wihtout issues.